src/etos_api/routers/etos/utilities.py

```diff
@@ -18,34 +18,49 @@
 import asyncio
 import time
 from etos_api.library.graphql import GraphqlQueryHandler
-from etos_api.library.graphql_queries import ARTIFACT_QUERY
+from etos_api.library.graphql_queries import (
+    ARTIFACT_IDENTITY_QUERY,
+    VERIFY_ARTIFACT_ID_EXISTS,
+)
 
 LOGGER = logging.getLogger(__name__)
 
 
-async def wait_for_artifact_created(etos_library, artifact_identity, timeout=30):
+async def wait_for_artifact_created(
+    etos_library, artifact_identity, artifact_id, timeout=30
+):
     """Execute graphql query and wait for an artifact created.
 
     :param etos_library: ETOS library instande.
     :type etos_library: :obj:`etos_lib.etos.ETOS`
     :param artifact_identity: Identity of the artifact to get.
     :type artifact_identity: str
+    :param artifact_id: ID of the artifact to get.
+    :type artifact_id: UUID
     :param timeout: Maximum time to wait for a response (seconds).
     :type timeout: int
     :return: ArtifactCreated edges from GraphQL.
     :rtype: list
     """
     timeout = time.time() + timeout
     query_handler = GraphqlQueryHandler(etos_library)
+    if artifact_id is not None:
+        LOGGER.info("Verify that artifact ID %r exists.", artifact_id)
+        query = VERIFY_ARTIFACT_ID_EXISTS
+    else:
+        LOGGER.info("Getting artifact from packageURL %r", artifact_identity)
+        query = ARTIFACT_IDENTITY_QUERY
+    identity = artifact_identity or str(artifact_id)
```

Comment:
Identity could easily be mixed up with artifact identity.
```suggestion
    artifact_identifier = artifact_identity or str(artifact_id)
```